utils/workers/Collectors.py

```diff
@@ -0,0 +1,161 @@
+
+"""
+Collectors - 
+Process 2 of 4
+These are designed to pull roi information from flywheel.
+The ROI information can be on the session level (for dicoms) or the file level (niftis)
+They must identify where the information is, and then in the case of
+session level dicom data, they will only pull out ROI's that are associated with
+the input file provided to the gear. 
+
+Responsibilities:
+1. locate ohif metadata
+2. isolate ROI's associated with input file
+3. return curated metadata
+    
+Full process:
+1. Prep
+2. Collect
+3. Generate
+4. Convert
+
+"""
+
+class CollWorker(ABC):
+    def __init__(self, fw_client, file_object, orig_dir):
+        self.fw_client = fw_client
+        self.orig_dir = orig_dir
+        self.file_object = file_object
+        self.ohifviewer_info = {}
+        self.validROIs = ["RectangleRoi", "EllipticalRoi", "FreehandRoi"]
+
+    def get_ohif_info(self):
+        # Can assume file is reloaded already as the file object does this automatically
+        flywheel_file = self.file_object.flywheel_file
+        print(flywheel_file.name)
+        if flywheel_file.get("info") and flywheel_file["info"].get("ohifViewer"):
+            self.ohifviewer_info = flywheel_file["info"].get("ohifViewer")
+            print('info on file')
+
+        else:
+            # session stores the OHIF annotations
+            print('info on session')
+            session = self.fw_client.get_session(flywheel_file["parents"]["session"])
+            print(session.label)
+            self.ohifviewer_info = session.info.get("ohifViewer")
+
+        if not self.ohifviewer_info:
+            error_message = "Session info is missing ROI data for selected DICOM file."
+            raise InvalidROIError(error_message)
+
+
+
+    @abstractmethod
+    def collect(self):
+        pass
+
+
+
+@dataclass
+class LabelCollector:
+    fw_client: Client
+    orig_dir: Path
+    file_object: FileEntry
+    collector: CollWorker = None
+
+    def __post_init__(self):
+        self.collector = self.collector(self.fw_client, self.file_object, self.orig_dir)
+
+    def collect(self):
+        return self.collector.collect()
+
+
+class DicomRoiCollector(CollWorker):
+    def collect(self):
+        self.get_ohif_info()
+        studyUID, seriesUID = self.get_current_study_series_uid()
+        self.identify_rois_on_image(studyUID, seriesUID)
+
+        return self.ohifviewer_info
+
+    def get_one_dicom(self):
+        dicom = None
+        for root, _, files in os.walk(str(self.orig_dir), topdown=False):
+            for fl in files:
+                try:
+                    dicom_path = Path(root) / fl
+                    dicom = pydicom.read_file(dicom_path, force=True)
+                    break
+                except Exception as e:
+                    log.warning("Could not open dicom file. Trying another.")
+                    log.exception(e)
+                    pass
+            if dicom:
+                break
+
+        return dicom
```

Comment:
I would also use `fw-file` for this if you can.  For example if you simply load the whole collection, you get these values from across the collection:

```python
from fw_file.dicom import DICOMCollection

dcms = DICOMCollection.from_dir(self.orig_dir)
studyInstance = dcms.get('StudyInstanceUID') # Get's value across the collection, raises a ValueError if multiple are found
seriesInstance = dcms.get('SeriesInstanceUID')